appengine-java-logging/src/test/java/com/google/apphosting/logging/JsonFormatterTest.java

```diff
@@ -85,19 +104,155 @@ public void messageIncludesStackTrace() throws Exception {
     BufferedReader reader = new BufferedReader(new StringReader(message));
     assertEquals("logger: message", reader.readLine());
     assertEquals("java.lang.Throwable: thrown", reader.readLine());
-    assertTrue(
-        reader
-            .readLine()
-            .startsWith("\tat " + getClass().getName() + ".messageIncludesStackTrace"));
+    assertTrue(reader.readLine()
+        .startsWith("\tat " + getClass().getName() + ".messageIncludesStackTrace"));
+  }
+
+  @Test
+  public void testCommonsLogging() {
+    final Instant now = Instant.now();
+    final String expectedThreadName = Thread.currentThread().getName();
+
+    // Generate Events from Commons Logging
+    new CommonsLoggingExample().run();
+
+    // Verify that event was captured by slf4j -> java.util.logging
+    String[][] expected = new String[][] {{"DEBUG", "A CommonsLogging Debug Event: "},
+        {"INFO", "A CommonsLogging Info Event: "}, {"WARNING", "A CommonsLogging Warn Event: "},
+        {"ERROR", "A CommonsLogging Error Event: "}};
+
+    List<String> events = capturingHandler.getEvents();
```

Comment:
I am not sure what you mean by "Each logger has different levels as well". I am not suggesting to eliminate the different `expected` arrays but rather have these tests:
```
assertThat("severity", data.severity, is(expected[i][0]));
assertThat("timestamp.seconds", data.timestamp.seconds, ...
assertThat("timestamp.nanos", data.timestamp.nanos,
assertThat("thread name", data.thread, is(expectedThreadName));
assertThat("logger message", data.message, ..
if (data.severity.equals("ERROR")) {
  ...
}
```

which seems to be the same factored into one method that will receive `expected` and `loggerMessagePrefix` as params.